pkg/service/service.go

```diff
@@ -679,10 +683,58 @@ func isRequired(required []string, name string) bool {
 
 // IsCSVSupported checks if the cluster supports resources of type ClusterServiceVersion
 func IsCSVSupported() (bool, error) {
-	occlient, err := occlient.New()
+	client, err := occlient.New()
 	if err != nil {
 		return false, err
 	}
 
-	return occlient.GetKubeClient().IsCSVSupported()
+	return client.GetKubeClient().IsCSVSupported()
+}
+
+// AddKubernetesComponentToDevfile adds service definition to devfile as an inlined Kubernetes component
+func AddKubernetesComponentToDevfile(crd, name string, devfileObj parser.DevfileObj) error {
+	err := devfileObj.Data.AddComponents([]devfile.Component{{
+		Name: name,
+		ComponentUnion: devfile.ComponentUnion{
+			Kubernetes: &devfile.KubernetesComponent{
+				K8sLikeComponent: devfile.K8sLikeComponent{
+					BaseComponent: devfile.BaseComponent{},
+					K8sLikeComponentLocation: devfile.K8sLikeComponentLocation{
+						Inlined: crd,
+					},
+				},
+			},
+		},
+	}})
+	if err != nil {
+		return err
+	}
+
+	return devfileObj.WriteYamlDevfile()
```

Comment:
Shouldn't this call be outside of this function?
This pattern seems a little bit weird to me. Based on the description and function name I would not expect to write the information directly to the file. What If I want to do more modification to the devfile structure and then write it all at once?

It might be better if the caller of this function can decide when to write devfile to file.

Reply:
> Shouldn't this call be outside of this function?

I don't have a strong opinion.

> This pattern seems a little bit weird to me.

I'm following the same approach as in:

* https://github.com/openshift/odo/blob/ba7161940c735caffaa9bd49e731dd15b8e9a61f/pkg/envinfo/url.go#L319-L333
* https://github.com/openshift/odo/blob/ba7161940c735caffaa9bd49e731dd15b8e9a61f/pkg/envinfo/url.go#L335-L360
* https://github.com/openshift/odo/blob/ba7161940c735caffaa9bd49e731dd15b8e9a61f/pkg/envinfo/storage.go#L120-L132
* https://github.com/openshift/odo/blob/ba7161940c735caffaa9bd49e731dd15b8e9a61f/pkg/envinfo/storage.go#L56-L78

> Based on the description and function name I would not expect to write the information directly to the file. 

Well the function comment does say that we put the info in devfile. I'm not sure how it's not saying what it's doing.

> What If I want to do more modification to the devfile structure and then write it all at once?

Would that modification be about service create and in the same odo command?

> It might be better if the caller of this function can decide when to write devfile to file.

Again, no strong opinion myself. Even if this is the right thing to do, I'm actually tired of accommodating so many changes that I have been making on this PR. I request to open a new issue for this.

Reply:
It is ok, it looks like we have made this mistake already in a lot of places. This will need to be cleaned up. But obviously not in this PR

Reply:
https://github.com/openshift/odo/issues/4607